python/private/hermetic_runtime_repo_setup.bzl

```diff
@@ -96,14 +103,65 @@ def define_hermetic_runtime_toolchain_impl(
         hdrs = [":includes"],
         includes = [
             "include",
-            "include/python{major}.{minor}".format(**version_dict),
-            "include/python{major}.{minor}m".format(**version_dict),
+        ] + select({
+            is_freethreaded: [
+                "include/python{major}.{minor}t".format(**version_dict),
+                # FIXME @aignas 2024-11-05: the following looks fishy - should
+                # we have a config setting for `m` (i.e. DEBUG builds)?
+                "include/python{major}.{minor}tm".format(**version_dict),
+            ],
+            "//conditions:default": [
+                "include/python{major}.{minor}".format(**version_dict),
+                "include/python{major}.{minor}m".format(**version_dict),
+            ],
+        }),
+    )
+    native.config_setting(
+        name = "is_freethreaded_linux",
+        flag_values = {
+            Label("//python/config_settings:py_freethreaded"): "yes",
+        },
+        constraint_values = [
+            "@platforms//os:linux",
+        ],
+        visibility = ["//visibility:private"],
+    )
+    native.config_setting(
+        name = "is_freethreaded_osx",
+        flag_values = {
+            Label("//python/config_settings:py_freethreaded"): "yes",
+        },
+        constraint_values = [
+            "@platforms//os:osx",
+        ],
+        visibility = ["//visibility:private"],
+    )
+    native.config_setting(
+        name = "is_freethreaded_windows",
+        flag_values = {
+            Label("//python/config_settings:py_freethreaded"): "yes",
+        },
+        constraint_values = [
+            "@platforms//os:windows",
```

Comment:
Discuss: should this go to somewhere outside the macro definition? Maybe we could have `//python/private/config_settings:BUILD.bazel` where such things can go? The downside is that the visibility will not be `private` anymore.

Reply:
I think its fine to locally define some config settings like this. These are fairly specific and not particularly complicated.